### ISSUE TITLE
fix(blanks-around-fences): single-line HTML comment before fence incorrectly triggers blank-line violation

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#331]
## Overview

`blanks-around-fences` incorrectly flags a fenced code block as missing a preceding blank line when the line immediately before the fence is a single-line HTML comment (e.g. a `<!-- gomarklint-disable-next-line -->` directive).

## Problem

`stepHTMLComment` only skips lines that **open** a multi-line HTML comment (`<!--` without `-->`). A self-contained single-line comment (`<!-- ... -->`) returns `skip=false`, so `prevBlank` is set to `false` and the opening fence on the very next line fires a false positive.

This creates a catch-22 with `disable-next-line`:

- Without a blank line → `blanks-around-fences` fires on the fence.
- With a blank line → the blank line becomes the "next line", so the disable directive no longer applies to the fence.

HTML comments are invisible in rendered Markdown, so they should not be treated as non-blank content for the purposes of this rule.

## Examples

| Input | Expected | Actual |
|---|---|---|
| `text\n\n<!-- gomarklint-disable-next-line -->\n` ``` `` ` ``go\n ``` | no violation | violation on fence line |
| `text\n\n<!--\ncomment\n-->\n` ``` `` ` ``go\n ``` | no violation | no violation (multi-line already skipped) |

## Fix

In `CheckBlanksAroundFences` (`internal/rule/blanks_around_fences.go`), update the HTML-comment handling so that:

1. **Single-line HTML comments** (`<!--` and `-->` on the same line) are recognized and skipped — `stepHTMLComment` should return `(true, false)` in this case.
2. When a line is skipped as an HTML comment, **`prevBlank` is preserved** (not forced to `false`), because the comment is invisible in rendered output and should not break a blank-line chain established before it.

## References

- `internal/rule/blanks_around_fences.go` — `CheckBlanksAroundFences`, `stepHTMLComment`
- `internal/linter/disable_comment.go` — `disable-next-line` directive parsing

**Comment by shinagawa-web** [comment]
## Implementation Plan

### Files to modify

- `internal/rule/blanks_around_fences.go`
  - `stepHTMLComment`: add a branch for single-line comments (`<!--` and `-->` on the same line) → return `(true, false)`
  - `CheckBlanksAroundFences`: when a comment line is skipped, preserve `prevBlank` instead of forcing it to `false`

### Files to modify (tests)

- `internal/rule/blanks_around_fences_test.go`
  - Add cases: single-line HTML comment preceded by blank → no violation
  - Add cases: single-line HTML comment preceded by non-blank → violation (prevBlank correctly carried through)
  - Add case: `disable-next-line` immediately before fence → no `blanks-around-fences` violation

### Checklist

- [ ] Update `stepHTMLComment` to handle single-line HTML comments
- [ ] Update `CheckBlanksAroundFences` to preserve `prevBlank` when skipping HTML comment lines
- [ ] Add / update unit tests in `blanks_around_fences_test.go`
- [ ] `make test-all` passes
